src/org.hdfgroup.object/hdf/object/Dataset.java

```diff
@@ -1591,6 +1591,35 @@ protected String toString(Object theData, Datatype theType, String delimiter, in
                 break;
             }
         }
+        else if (theType.isFloat() && theType.getDatatypeSize() == 2) {
+            Object value = Array.get(theData, 0);
+            String strValue;
+
+            if (value == null)
+                strValue = "null";
+            else
+                strValue = Float.toString(Float.float16ToFloat((short)value));
+
+            // if (count > 0 && strValue.length() > count)
+            // truncate the extra characters
+            // strValue = strValue.substring(0, count);
+            sb.append(strValue);
+
+            for (int i = 1; i < count; i++) {
+                sb.append(delimiter);
+                value = Array.get(theData, i);
+
+                if (value == null)
+                    strValue = "null";
+                else
+                    strValue = Float.toString(Float.float16ToFloat((short)value));
+
+                if (count > 0 && strValue.length() > count)
```

Comment:
Not sure what this line is trying to do. `count` must always be > 0 at this point and `count` seems to refer to some limit on the maximum number of items to return, so I'm not sure why the length of the string is being compared to `count` here, unless `count` is in terms of characters here. If that's the case, it also seems like the `sb.append` call above needs to account for `count`, but that code's commented out. However, I think this whole thing can be simplified into just a loop that starts at index 0, rather than getting a single element and then having a loop to look for other elements.

In fact, it seems like all the other cases above can and should be simplified the same way, but that's completely separate to this PR.

Reply:
Can you convert this comment to an issue?

Reply:
Probably a cut-n-paste issue

Reply:
Looks like that is copied from the else clause, which also had that commented out.